### PR TITLE
Feedback app

### PR DESCRIPTION
**Pull request by prashant2796**

### CODE REVIEW

File: feedback/admin.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+class FeedbackModelAdmin(admin.ModelAdmin):
+ 	"""
+	handles the admin panel for feedback model.
+ 	"""
+ 	list_display = ['user', 'text', 'is_active', 'created_at', 'modified_at']
+ 	list_display_links = ['user']
+ 	search_fields = ['text']
+ 	autocomplete_fields = ['user']
+
```
Comment (Par010) [inline feedback/admin.py]: add a list_filter

File: feedback/api/permissions.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    message = "You must be the owner of the object to view."
+
+    def has_object_permission(self, request, view, obj):
+        if request.method in SAFE_METHODS:
+            if obj.user == None:
+                # if user is anonymous then read-only
+                return True
+
+            # Check permissions for read-only request
+            elif obj.user == request.user:
```
Comment (Par010) [inline feedback/api/permissions.py]: no need for custom permissions just use IsAuthenticated

File: feedback/api/views.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	def perform_update(self, serializer):
+		serializer.save(user=self.request.user)
+
+	def get_queryset(self):
+		user = self.request.user.id
+		# Get objects only associated with the requesting user
+		queryset = Feedback.objects.filter(user=user)
+		return queryset
+
+
```
Comment (Par010) [inline feedback/api/views.py]: no need for an endpoint with user-specific feedbacks

File: feedback/api/permissions.py (hunk 2 above)
Comment (darth-dodo) [reply feedback/api/permissions.py]: why is this branch merged? this is not fixed

File: feedback/api/views.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	def create(self, request, *args, **kwargs):
+		serializer = self.get_serializer(data=request.data)
+		serializer.is_valid(raise_exception=True)
+		self.perform_create(serializer)
+		headers = self.get_success_headers(serializer.data)
+		return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
+
+	def perform_create(self, serializer):
+		if self.request.user.id == None:
```
Comment (darth-dodo) [inline feedback/api/views.py]: How does the `request.user` look for an Anonymous user?

File: feedback/models.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+from django.contrib.auth.models import User
+from courses.utils.modelsutils import RowInformation
+
+# Create your models here.
+
+class Feedback(RowInformation):
+	"""
+	class to store feedback information.
+	"""
+	user = models.ForeignKey(User, null=True, blank=True, on_delete=models.DO_NOTHING)
```
Comment (darth-dodo) [inline feedback/models.py]: good job on `on_delete` 😄

File: feedback/api/views.py (hunk 4 above)
Comment (prashant2796) [reply feedback/api/views.py]: If there is an Anonymous user then the value for user column would be Null.
i.e there will be no id assigned to it.
So this perform create method first checks if there is a user with no id i.e 
"""
if self.request.user.id == None
"""
if there is no value in the user column then save the serializer with Null as the value for user column else save the user with id. 
If you test the code you will see that an anonymous 
user's object, the value against user column is null

File: feedback/api/permissions.py (hunk 2 above)
Comment (prashant2796) [reply feedback/api/permissions.py]: Actually, with this method, I am not allowing any type of users to put or patch.
Is it not the correct way to do it?

Comment (darth-dodo) [reply feedback/api/permissions.py]: explore this http://www.django-rest-framework.org/api-guide/generic-views/#listcreateapiview

Also, return only that user specific data in retrieving

File: feedback/api/views.py (hunk 4 above)
Comment (darth-dodo) [reply feedback/api/views.py]: No. What comes in the request. print out on the console and share it.

`print request.user.__dict__` for anonymous requests

File: feedback/api/permissions.py (hunk 2 above)
Comment (prashant2796) [reply feedback/api/permissions.py]: hey made this changes and pushed the new branch with PR raised.
https://github.com/projectlumos/project-lumos-backend/pull/37

File: feedback/api/views.py (hunk 4 above)
Comment (prashant2796) [reply feedback/api/views.py]: if I print #print(self.request.user.id) when any user is logged in it prints user's id in the console
and in the case of the anonymous user, it prints None.
But if I print #print(self.request.user.__dict__)  when any user is logged in it prints something like
{'_setupfunc': <function AuthenticationMiddleware.process_request.<locals>.<lambda> at 0x000001EA2E879950>, '_wrapped': <User: psp>} 
and in the case of anonymous user it prints out empty dict.

Comment (prashant2796) [reply feedback/api/views.py]: I think that's because the request is made by the user who is not stored in the database

**Pull request merged**